packages/workspace/src/workspace/nacl_files/nacl_files_source.ts

```diff
@@ -101,9 +102,11 @@ type NaclFilesState = {
   readonly referencedIndex: Record<string, string[]>
 }
 
-const cacheResultKey = (filename: string, timestamp?: number): ParseResultKey => ({
+const cacheResultKey = (filename: string, buffer: ContentType,
```

Comment:
nit: maybe make this function get the naclFile instead of getting each member separately?
if we only need specific keys we can define this as
```suggestion
const cacheResultKey = (naclFile: Pick<NaclFile, 'filename' | 'buffer' | 'timestamp'>): ParseResultKey => ({
```
but it would probably be just as good to simply get the whole NaclFile
```suggestion
const cacheResultKey = (naclFile: NaclFile): ParseResultKey => ({
```